ouroboros-consensus-cardano/tools/db-validator/Class.hs

```diff
@@ -1,11 +1,17 @@
-{-# LANGUAGE FlexibleInstances   #-}
-{-# LANGUAGE NamedFieldPuns      #-}
-{-# LANGUAGE OverloadedStrings   #-}
-{-# LANGUAGE RecordWildCards     #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TypeApplications    #-}
-
-module Analysis (HasAnalysis (..), mkTopLevelConfig) where
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE NamedFieldPuns         #-}
+{-# LANGUAGE OverloadedStrings      #-}
+{-# LANGUAGE RecordWildCards        #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeApplications       #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+
+module Class (
```

Comment:
I'm not a fan of the name `Class`, it's rather vague, it contains type classes, I suppose.

What if you spread the instances over `Block.Byron`, `Block.Shelley`, and `Block.Cardano`?